pkg/apis/extensions/register.go

```diff
@@ -47,6 +47,11 @@ var (
 
 // Adds the list of known types to api.Scheme.
 func addKnownTypes(scheme *runtime.Scheme) error {
+	// Add known resource names introduced in this API
+	api.AddStandardResourceNames(ResourceDeployments, ResourceReplicaSets)
```

Comment:
still looking through the PR, but from a code structure standpoint, it would be good if we could have done the same call patttern for the non-extension group.  so the no-name group...